python/dgl/heterograph.py

```diff
@@ -4799,6 +4799,95 @@ def multi_update_all(self, etype_dict, cross_reducer, apply_node_func=None):
             if apply_node_func is not None:
                 self.apply_nodes(apply_node_func, ALL, self.ntypes[dtid])
 
+    #################################################################
+    # New Message passing on heterograph // will replace *update_all
+    #################################################################
+    def update_all_new(self,
+                       message_func,
+                       reduce_func,
+                       apply_node_func=None):
```

Comment:
Can we merge this function into `update_all` in this release?  It's OK not supporting every combination of message and reduce functions for the case of heterogeneous graphs.